### PR TITLE
[release-v1.8] SRVKE-1229: Register schemas in webhooks too

### PR DESCRIPTION
**Pull request by creydr**
Cherry-pick of https://github.com/knative/eventing/pull/6633

Jira: [SRVKE-1229](https://issues.redhat.com/browse/SRVKE-1229)

### CODE REVIEW

File: config/core/roles/webhook-clusterrole.yaml [hunk 1]
```diff
@@ -98,6 +98,18 @@ rules:
       - "leases"
     verbs: *everything
 
+  # For creating events
+  - apiGroups:
```
Comment (matzew) [inline config/core/roles/webhook-clusterrole.yaml:102]: can you run the `make RELEASE.....` target on the make file?

this updates the midstream manifests

Comment (pierDipi) [reply config/core/roles/webhook-clusterrole.yaml:102]: the full command would be:
```
make generate-release RELEASE=v1.8
```

Comment (pierDipi) [reply config/core/roles/webhook-clusterrole.yaml:102]: No, it's that we have inconsistency and every repo accepts it's own "RELEASE" format

File: openshift/release/artifacts/eventing-core.yaml [hunk 2]
```diff
@@ -4825,45 +4837,3 @@ spec:
...
-  name: openshift-serverless-view-eventing-configmaps
-  namespace: knative-eventing
-subjects:
-  - kind: Group
-    name: system:authenticated
-    apiGroup: rbac.authorization.k8s.io
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: Role
-  name: openshift-serverless-view-eventing-configmaps
```
[the remark this answers is not in the document]
Comment (matzew) [reply openshift/release/artifacts/eventing-core.yaml:4869]: We had done something about this ...

[the remark this answers is not in the document]
Comment (creydr) [reply openshift/release/artifacts/eventing-core.yaml:4869]: seems we're only doing this in `create-release-branch.sh` and `update-to-head.sh`:
https://github.com/openshift-knative/eventing/blob/1852f8e8d2f9ebe5927b3e0491a0a6c0fed8f149/openshift/release/create-release-branch.sh#L27-L37

[the remark this answers is not in the document]
Comment (matzew) [reply openshift/release/artifacts/eventing-core.yaml:4869]: on the old repo we did someting, b/c the patch was never applied on the branch creation as well. See:
https://github.com/openshift/knative-eventing/pull/1931/files


/hold

I am looking _now_ what's missing

File: config/core/roles/webhook-clusterrole.yaml (hunk 1 above)
Comment (matzew) [reply config/core/roles/webhook-clusterrole.yaml:102]: No, it is supposed to be `make RELEASE=v1.8`, the fact that you need to do `knative-v1.8.0` is IMO wrong.

Our "older" branches work just with `v1.6`, for instance:
https://github.com/openshift-knative/eventing/blob/release-v1.6/openshift/release/artifacts/eventing-core.yaml#L20
(and the matching SO repo: https://github.com/openshift-knative/serverless-operator/blob/main/openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.6/2-eventing-core.yaml#L20)

But for 1.8, this is different, on branch creation

Comment (pierDipi) [reply config/core/roles/webhook-clusterrole.yaml:102]: Yeah, the release param here: https://github.com/openshift-knative/eventing/blob/8e00ab32a7fcc02324590cfb6e2fb063786db137/openshift/release/generate-release.sh#L21

is also expecting only v1.8, so I'm not sure why it worked with the knative-v1.8.0

**Comment by matzew** [comment]
Thanks for the rebase!

/cherry-pick release-v1.7

**Pull request merged**